test/cache.ts

```diff
@@ -44,7 +44,7 @@ test('cached response is re-encoded to current encoding option', withServer, asy
 	const firstEncoding = 'base64';
 	const secondEncoding = 'hex';
 
-	const firstResponse = await got({cache, encoding: firstEncoding});
+	const firstResponse = await got<string>({cache, encoding: firstEncoding});
```

Comment:
`got` by default returns a string, so I think it should automatically infer a `string` here. It should also have overloads to change the type depending on the `responseType` option.

Reply:
Updated in types and tests.